src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingService.java

```diff
@@ -291,6 +292,9 @@ private boolean isAllowedWifiConnected() {
     private void updateNotification() {
         SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(this);
         String type = sp.getString(PREF_NOTIFICATION_TYPE, "low_priority");
+        boolean foreground = sp.getBoolean(PREF_FOREGROUND_SERVICE, false);
+        if ("none".equals(type) && foreground)
+            type = "low_priority"; // foreground priority requires a notification
```

Comment:
This whole thing looks really complicated with changing the value of foreground. I think you can just get rid of this and move `builder.setPriority(NotificationCompat.PRIORITY_MIN);` to line 312.